x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -2263,4 +2269,30 @@ private static AggregateMetricDoubleBlockBuilder.Metric getMetric(AggregateFunct
             return null;
         }
     }
+
+    private static class ResolveFullTextSearch extends ParameterizedAnalyzerRule<Filter, AnalyzerContext> {
+        @Override
+        protected LogicalPlan rule(Filter filter, AnalyzerContext context) {
+            Expression condition = filter.condition();
+
+            // Replace every instance of FullTextSearch
+            if (condition instanceof FullTextSearch fts) {
```

Comment:
Alternatively we could check, if `MultiMatch` contains one field, which is an instance of `UnresolvedStar` instead of introducing a separate "marker class" `FullTextSearch`.